### PR TITLE
pkg/asset/ignition/bootstrap/cvoignore: Add group/kind/name(space) collision detection

### PR DESCRIPTION
**Pull request by wking**
[`docs/design/resource_dep.svg`][1] shows the complicated hierarchy of installer assets, and it is likely that not all users are familiar with all the manifests that the installer generates.  It's also possible that there is a complicated hierarchy on the caller's side, and not just a single actor, creating additional manifests to feed into a create-manifests flow.  With all of these inputs, it seems possible that folks occasionally call the installer with directory contents that result in multiple manifests for a single resource, as determined by the (group, kind, namespace, name) tuples.  This commit adds a check at the slightly-before-Ignition-serialization overrides asset, which was already in the business of iterating over the content and building ClusterVersion spec.overrides.  Failing fast before we build Ignition configs should allow for relatively cheap recovery, vs. a user's cluster failing to come up, or coming up with a config that diverges from their intention because some other manifest ended up taking precedence over the resource that they'd been trying to control.

Not covered in this commit is deduping between manifests generated on the bootstrap machine (e.g. by the various 'render' calls to operator containers).  But that can happen orthogonally, or not, and I think it's still worth having this cheap, pre-Ignition-config sanity check.

[1]: https://github.com/openshift/installer/blob/b08fe18f02768c2db33b043eba4925377b0b6cb7/docs/design/resource_dep.svg

### CODE REVIEW

**Comment by wking** [comment]
Lots [of][1] `Init container ... not ready`.

/retest

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_installer/6247/pull-ci-openshift-installer-master-gofmt/1560888165072375808

File: pkg/asset/ignition/bootstrap/cvoignore.go [hunk 1]
```diff
@@ -51,21 +52,34 @@ func (a *CVOIgnore) Generate(dependencies asset.Parents) error {
...
 		u := &unstructured.Unstructured{}
 		if err := yaml.Unmarshal(file.Data, u); err != nil {
 			return errors.Wrapf(err, "could not unmarshal %q", file.Filename)
 		}
+		group := u.GetObjectKind().GroupVersionKind().Group
+		kind := u.GetKind()
+		namespace := u.GetNamespace()
+		name := u.GetName()
+
+		key := fmt.Sprintf("%s |! %s |! %s |! %s", group, kind, namespace, name)
```
Comment (patrickdillon) [inline pkg/asset/ignition/bootstrap/cvoignore.go:67]: Out of curiosity, why the `|!` separator?

**Review by patrickdillon — COMMENTED** [review]
/lgtm
/assign @jhixson74 
/retest

File: pkg/asset/ignition/bootstrap/cvoignore.go (hunk 1 above)
Comment (wking) [reply pkg/asset/ignition/bootstrap/cvoignore.go:67]: I didn't spend much time poking around in allowed-characters for the four entries, and I wanted a delimiter that was unlikely to show up in any of them.

Comment (wking) [reply pkg/asset/ignition/bootstrap/cvoignore.go:67]: it would perhaps be more robust to just use `%q%q%q%q`?

**Comment by jhixson74** [comment]
This looks straight forward to me, I just want to see the tests pass first ;-)

**Comment by wking** [comment]
[Confirmed][1] that this works as expected, with an error message that could be polished further, but which is already good enough to satisfy me ;)

[1]: https://github.com/openshift/release/pull/32306#issuecomment-1247636876

**Pull request merged**